src/main/java/org/spongepowered/common/data/processor/data/entity/IgniteableDataProcessor.java

```diff
@@ -0,0 +1,147 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class IgniteableDataProcessor extends AbstractSpongeDataProcessor<IgniteableData, ImmutableIgniteableData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof net.minecraft.entity.Entity;
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public Optional<IgniteableData> from(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            final SpongeIgniteableData igniteableData = new SpongeIgniteableData();
+            final net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) (dataHolder);
+            return Optional.<IgniteableData>of(igniteableData.setFireTicks(entity.fire).setFireDelay(entity.fireResistance));
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<IgniteableData> fill(DataHolder dataHolder, IgniteableData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final IgniteableData merged = overlap.merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+            manipulator.set(Keys.FIRE_DAMAGE_DELAY, merged.fireDelay().get()).set(Keys.FIRE_TICKS, merged.fireTicks().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<IgniteableData> fill(DataContainer container, IgniteableData igniteableData) {
+        igniteableData.set(Keys.FIRE_TICKS, getData(container, Keys.FIRE_TICKS));
+        igniteableData.set(Keys.FIRE_DAMAGE_DELAY, getData(container, Keys.FIRE_DAMAGE_DELAY));
+        return Optional.of(igniteableData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, IgniteableData manipulator, MergeFunction function) {
+        final int oldFireDamageDelay = manipulator.fireDelay().get();
+        final int oldFireTicks = manipulator.fireTicks().get();
+        final IgniteableData merged =
+                checkNotNull(function, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+        manipulator.set(Keys.FIRE_DAMAGE_DELAY, merged.fireDelay().get());
+        manipulator.set(Keys.FIRE_TICKS, merged.fireTicks().get());
+        final int fireDamageDelay = merged.fireDelay().get();
+        final int fireTicks = merged.fireTicks().get();
+        if (this.supports(dataHolder)) {
+            net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) dataHolder;
+            entity.fire = fireTicks;
+            entity.fireResistance = fireDamageDelay;
+            Collection<ImmutableValue<?>> successful = new ArrayList<ImmutableValue<?>>();
+            Collection<ImmutableValue<?>> replaced = new ArrayList<ImmutableValue<?>>();
+            successful.add(new ImmutableSpongeValue<Integer>(Keys.FIRE_TICKS, oldFireTicks));
```

Comment:
You're using the wrong type of value here. It's supposed to be an `ImmutableSpongeBoundedValue`.